apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -1,18 +1,66 @@
 import * as express from 'express';
 import asyncHandler from 'express-async-handler';
 import { OpenAI } from 'openai';
+import { z } from 'zod';
 
 import * as error from '@prairielearn/error';
+import { loadSqlEquiv, queryRow } from '@prairielearn/postgres';
 
 import { config } from '../../../lib/config.js';
+import { QuestionSchema, IdSchema } from '../../../lib/db-types.js';
+import { QuestionAddEditor } from '../../../lib/editors.js';
 import { features } from '../../../lib/features/index.js';
-import { generateQuestion } from '../../lib/aiQuestionGeneration.js';
+import { idsEqual } from '../../../lib/id.js';
+import { selectJobsByJobSequenceId } from '../../../lib/server-jobs.js';
+import { generateQuestion, regenerateQuestion } from '../../lib/aiQuestionGeneration.js';
 import { syncContextDocuments } from '../../lib/contextEmbeddings.js';
 
-import { AiGeneratePage } from './instructorAiGenerateQuestion.html.js';
+import { AiGeneratePage, GenerationResults } from './instructorAiGenerateQuestion.html.js';
+
+type IdType = z.infer<typeof IdSchema>;
```

Comment:
No need for this, you can use `string` directly.

Reply:
done in e0331beaf9d587990dcda087d800feedc786fc87.